### DISCUSSION TITLE
Feature request: Ability to swap column sizes...

### DISCUSSION THREAD

**Discussion by minego** [YaLTeR/niri#393] [Ideas]
I've used things like DWM, awesomewm, DWL, etc for years, and one thing I miss from those is that I can select a window and hit mod+enter and that swaps with the "master"...

I think I'd be able to replicate that use case really nicely in niri if there was a way to quickly swap the width of the current column with the column to the left or right of it.

So, if I have two columns, one that is 2/3 the screen width and the other 1/3, then swap focus but swap the widths as well.

One approach that I think would work well would be a modified version of `move-column-left` and `move-column-right` that swaps the widths as well when moving. Then I can setup a "wide" column and a skinny column and move windows in and out of that as needed.

Thoughts? I'll probably start working on a pull request for this...

**Comment by YaLTeR** [comment]
Hm, you can kind of do it manually with a script doing `niri msg action set-column-width 66%` commands.
